### PR TITLE
Move out const strings in pkg/scheduler/api/well_known_labels.go

### PR DESCRIPTION
**Pull request by Huang-Wei**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

- Move out const strings in `pkg/scheduler/api/well_known_labels.go` to `pkg/apis/core/taint_constants.go`.
- A follow up of #83578 so as to eliminate package `pkg/scheduler/api`.

Please note that 

**Which issue(s) this PR fixes**:

Part of #82390.

**Special notes for your reviewer**:

- `TaintExternalCloudProvider` and `TaintNodeShutdown` are left in `pkg/scheduler/api/well_known_labels.go`. They will be taken care of by #84364. cc/ @yastij 
- If it's needed to expose the const strings externally, we can duplicate them into `staging/src/k8s.io/api/core/v1/` (not included yet in this PR)

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

/sig scheduling
/assign @ahg-g @liggitt @dims

### CODE REVIEW

**Review by liggitt — COMMENTED** [review]
> * If it's needed to expose the const strings externally, we can duplicate them into `staging/src/k8s.io/api/core/v1/` (not included yet in this PR)

in general, label keys expected to be used for cross component communication are appropriate for inclusion in the external API packages.

I would actually question why we need the internal constants at all.

**Comment by Huang-Wei** [comment]
@liggitt Comments addressed. PTAL.

File: staging/src/k8s.io/api/core/v1/well_known_labels.go [hunk 1]
```diff
@@ -39,4 +39,40 @@ const (
 	// controllers and kube-proxy to check if the Endpoint objects should be replicated when
 	// using Headless Services
 	IsHeadlessService = "service.kubernetes.io/headless"
+
+	// TaintNodeNotReady will be added when node is not ready
+	// and feature-gate for TaintBasedEvictions flag is enabled,
+	// and removed when node becomes ready.
+	TaintNodeNotReady = "node.kubernetes.io/not-ready"
```
Comment (liggitt) [inline staging/src/k8s.io/api/core/v1/well_known_labels.go]: package location looks good, but I'd put these in a file named `well_known_taints.go`

Comment (Huang-Wei) [reply staging/src/k8s.io/api/core/v1/well_known_labels.go]: done.

**Pull request merged**